src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -249,6 +252,17 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, BindEntry bindE
                 format = GetAttributeContent(bindEntry.BindFormatNode);
             }
 
+            if (TryGetFormatNode(
```

Comment:
Should we separate these into a separate pass to make it easy to delete in the future? I'm a little worried to see munging on the happy path, but I could be convinced otherwise.

Reply:
Technically this pass removes all the extra nodes like these and rewrites to ComponentAttributes. While this case is easy to do in a separate pass, teh diagnostic for `bind-value-onchange` will be hard to add later on because it would already be rewritten.
Also, when we want to delete these diagnostics, we also want to cleanup `BindTagHelperDescriptorProvider` to remove the bound attribute descriptors for `format-` attributes. So a separate pass here won't contain all the temporary code.